api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -900,41 +900,34 @@ message Cluster {
   // :ref:`STRICT_DNS<envoy_v3_api_enum_value_config.cluster.v3.Cluster.DiscoveryType.STRICT_DNS>`
   // and :ref:`LOGICAL_DNS<envoy_v3_api_enum_value_config.cluster.v3.Cluster.DiscoveryType.LOGICAL_DNS>`
   // this setting is ignored.
-  // Setting this value causes failure if the
-  // ``envoy.restart_features.use_apple_api_for_dns_lookups`` runtime value is true during
-  // server startup. Apple's API only allows overriding DNS resolvers via system settings.
   // This field is deprecated in favor of *dns_resolution_config*
   // which aggregates all of the DNS resolver configuration in a single message.
   repeated core.v3.Address dns_resolvers = 18
       [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
 
   // Always use TCP queries instead of UDP queries for DNS lookups.
-  // Setting this value causes failure if the
-  // ``envoy.restart_features.use_apple_api_for_dns_lookups`` runtime value is true during
-  // server startup. Apple' API only uses UDP for DNS resolution.
```

Comment:
How come these caveats are dropped in docs?

Reply:
use_tcp_for_dns_lookups  setting is for c-ares.  
With the Apple DNS resolver added by https://github.com/envoyproxy/envoy/pull/13074/, it added some configurations check cross different DNS resolver types:  https://github.com/envoyproxy/envoy/blob/7ec3032516827abed23fbc8e4e0ff62449ee8fc9/source/common/event/dispatcher_impl.cc#L168.

With this refactoring PR, these cross checks are removed. For example, if Apple DNS resolver is used(either by typed_dns_resolver_config, or by run-time flag+MacOS), it will use UDP anyway. Those c-ares specific DNS configurations (dns_resolvers, use_tcp_for_dns_lookups, dns_resolution_config), if present, are simply ignored.  IMHO, crossing checking these configuration fields are unnecessary. Also, those fields are deprecated fields, and there are warnings for them already.